storage/src/vespa/storage/persistence/persistencethread.cpp

```diff
@@ -277,6 +280,22 @@ spi::ReadConsistency api_read_consistency_to_spi(api::InternalReadConsistency co
     }
 }
 
+document::FieldSet::SP
+getFieldSet(const document::FieldSetRepo & repo, vespalib::stringref name, MessageTracker & tracker) {
+    try {
+        return repo.getFieldSet(name);
+    } catch (document::FieldNotFoundException & e) {
+        tracker.fail(storage::api::ReturnCode::ILLEGAL_PARAMETERS,
+                     make_string("Field %s not found in fieldset %s",
```

Comment:
this sounds like the field is not found in the field set, when the field is already in the field set, but not found in the document.